File: backend/src/messages/StatusMessage.ts
import { UUID } from "crypto";
import { MessageType } from "./MessageType";
import { StatusMessageData } from "./StatusMessageData";
import BaseMessage from "./BaseMessage";

export default class StatusMessage extends BaseMessage {
    
    /** Tipo del mensaje, en este caso, un estado. */
    type: MessageType = MessageType.STATUS;
    
    /** Contenido del mensaje, incluyendo la pregunta y el historial de conversación. */
    content: StatusMessageData = { state: ""};

    /**
     * Crea una nueva instancia de StatusMessage.
     * @param id - Identificador único del mensaje.
     * @param content - Contenido del mensaje con la pregunta y su historial.
     */
    constructor(id: UUID, content: StatusMessageData) {
        super(id);
        this.content = content;
    }

    /**
     * Obtiene el identificador del mensaje.
     * @returns El UUID del mensaje.
     */
    getId(): UUID {
        return this.id;
    }
    
    /**
     * Obtiene el tipo del mensaje.
     * @returns El tipo del mensaje, en este caso, MessageType.STATUS.
     */
    getType(): MessageType {
        return this.type;
    }

    /**
     * Obtiene el contenido del mensaje.
     * @returns El contenido del mensaje, de tipo StatusMessageData.
     */
    getContent<T = StatusMessageData>(): T {
        return this.content as T;
    }
}

File: api-python/messages/StatusMessage.py
from pydantic import BaseModel
from typing import List
from uuid import UUID
from .BaseMessage import BaseMessage
from .MessageType import MessageType

class StatusMessageContent(BaseModel):
    """
    Contenido de un mensaje de estado.

    Esta clase define el contenido específico de un mensaje de tipo `StatusMessage`, 
    que incluye el estado actual del sistema o proceso.
    """

    state: str
    """
    Estado del sistema o proceso.

    Este campo almacena el estado actual del sistema o proceso que se está monitoreando, 
    como "CALCULANDO", "RESPONDIENDO", etc.
    """

class StatusMessage(BaseMessage[StatusMessageContent]):
    """
    Mensaje de estado.

    Esta clase extiende `BaseMessage` y agrega un contenido específico de estado. 
    Representa un mensaje que indica el estado de un proceso o sistema en curso.
    """

    def __init__(self, id: UUID, state: str):
        """
        Inicializa un mensaje de estado.

        Este constructor crea un mensaje de tipo `STATUS` con un identificador único y 
        un estado determinado.

        Args:
            id (UUID): El identificador único del mensaje de estado.
            state (str): El estado del proceso o sistema, como "CALCULANDO" o "RESPONDIENDO".
        """
        super().__init__(id=id, type=MessageType.STATUS, content=StatusMessageContent(state=state))
